### PR TITLE
Phoenix release 1: Add support for editing stopped entries

### LINKED ISSUE #1564: [Phoenix] Show the Edit view on clicking stopped entries
- Show the edit view from the Edit running entry
- Show it on clicking any stopped entry

### LINKED ISSUE #1565: [Phoenix] Allow editing of stopped entries
- Persist changes made in the edit view

### LINKED ISSUE #1579: [Phoenix] Allow expanding grouped entries
- Allow expanding grouped entries
- Click entry to expand grouped entry

### PR DESCRIPTION
**Pull request by shantanuraj**
Please remember the [Contributing Guidelines](https://github.com/toggl/toggl-button/blob/master/.github/CONTRIBUTING.md) :heart:

## :star2: What does this PR do?

- Adds support for editing stopped entries
- Support for changing start/stop time will come in a later release

## :bug: Recommendations for testing

All changes should be tested across Chrome and Firefox.

- Changes have already been tested in smaller PRs #1578, #1580 and #1582 
- So just verifiy if the Edit form works well and editing running and stopped entries works as expected

## :memo: Links to relevant issues or information

Closes #1564 
Closes #1565 
Closes #1579

### CODE REVIEW

File: src/scripts/components/TimeEntriesList.tsx [hunk 1]
```diff
@@ -93,41 +93,100 @@ type TimeEntriesListItemProps = {
...
+  const isBillable = !!timeEntry.billable;
+  const tags = (timeEntry.tags && timeEntry.tags.length > 0)
+    ? timeEntry.tags.join(', ')
+    : '';
+
+  return (
+    <EntryItem onClick={onRowClick}>
       <EntryItemRow>
-        {entriesCount > 1 &&
+        {renderHeader && entriesCount && earliestStartTime &&
```
Comment (tcrammond) [inline src/scripts/components/TimeEntriesList.tsx]: would be nice if all the conditions around grouping were a bit more concise, but not a big deal

Comment (shantanuraj) [reply src/scripts/components/TimeEntriesList.tsx]: It's just `renderHeader` that matters but I do an existential check here
so in the next line TS doesn't complain about optional properties

File: src/scripts/components/TimeEntriesList.tsx [hunk 2]
```diff
@@ -93,41 +93,100 @@ type TimeEntriesListItemProps = {
   project: Toggl.Project | null;
 };
 function TimeEntriesListItem ({ timeEntries, project }: TimeEntriesListItemProps) {
+  const [groupEntries, setGroupEntries] = React.useState(true)
```
[the remark this answers is not in the document]
Comment (shantanuraj) [reply src/scripts/components/TimeEntriesList.tsx]: Going with `isGroupeCollapsed` does that work?

[the remark this answers is not in the document]
Comment (tcrammond) [reply src/scripts/components/TimeEntriesList.tsx]: :doit:

**Comment by shantanuraj** [comment]
@tcrammond about the lagging billable toggle

Here's a video of me manually adding the checked class to the element and the DOM not refreshing
until I focus on it

https://toggl.slack.com/files/U8U0D02S3/FQDNYD5FC/nov-20-2019_23-29-16.mp4

I think it's related to how Chrome is throttling animation and timers oon extensions

**Comment by shantanuraj** [comment]
Issue for weird grouping bugs
https://github.com/toggl/toggl-button/issues/1588#issue-526083719

I did some investigation that suggests it maybe related to the `isSameDay` check on the start times
but with the debugger attached I couldn't get the popup into a weird  grouping state.

Have fixed other bugs :bow:

**Pull request merged**